common/src/main/java/com/wynntils/features/user/overlays/InfoBoxesFeature.java

```diff
@@ -0,0 +1,218 @@
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.objects.CustomColor;
+import com.wynntils.wynn.objects.EmeraldSymbols;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.resources.language.I18n;
+
+@FeatureInfo(category = FeatureCategory.OVERLAYS)
+public class InfoBoxesFeature extends UserFeature {
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox1Overlay = new InfoBoxOverlay(1);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox2Overlay = new InfoBoxOverlay(2);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox3Overlay = new InfoBoxOverlay(3);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox4Overlay = new InfoBoxOverlay(4);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox5Overlay = new InfoBoxOverlay(5);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox6Overlay = new InfoBoxOverlay(6);
+
+    private static final Pattern INFO_VARIABLE_PATTERN =
+            Pattern.compile("%([a-zA-Z_]+|%)%|\\\\([\\\\n%§EBLMH]|x[\\dA-Fa-f]{2}|u[\\dA-Fa-f]{4}|U[\\dA-Fa-f]{8})");
+
+    public static class InfoBoxOverlay extends Overlay {
+        @Config
+        public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.OUTLINE;
+
+        @Config
+        public String content = "";
+
+        private final int id;
+
+        private TextRenderTask toRender = recalculateFunctions(content);
+        private List<Function<?>> functionDependencies = new ArrayList<>();
+
+        public InfoBoxOverlay(int id) {
+            super(
+                    new OverlayPosition(
+                            -60 + (15 * id),
+                            5,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Left,
+                            OverlayPosition.AnchorSection.MiddleLeft),
+                    new GuiScaledOverlaySize(120, 10),
+                    HorizontalAlignment.Left,
+                    VerticalAlignment.Middle);
+            this.id = id;
+        }
+
+        private TextRenderTask recalculateFunctions(String renderableText) {
+            if (functionDependencies == null) {
+                functionDependencies = new ArrayList<>();
+            }
+
+            for (Function<?> oldDependency : functionDependencies) {
+                FunctionManager.disableFunction(oldDependency);
+            }
+
+            functionDependencies.clear();
+
+            StringBuilder builder = new StringBuilder(renderableText.length() + 10);
+            Matcher m = INFO_VARIABLE_PATTERN.matcher(renderableText);
```

Comment:
There is a `FunctionManager.doFormat()` which uses the new `{function}` syntax. But I see now that I never finished `getStringFromTemplate()`, which is that you would have needed here.

I am not entirely sure I want to import the old `%info` syntax. We can possibly do this short-term if you really want to push this now, as long as we agree that this is just temporary.

Reply:
I did not even know that was a thing. This is mostly code I took from DonkeyBlaster, and made it work. Yes, I remember all the new things you want to do with function variables, but most of that code right now is TODO. So, until that is changed, we should push this.